packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -118,6 +118,49 @@ export class ProfilesCache {
         }
     }
 
+    public async refreshConfig(apiRegister: ZoweExplorerApi.IApiRegisterClient): Promise<void> {
+        this.allProfiles = [];
+        let tmpAllProfiles = [];
+        this.allTypes = [];
+        const mProfileInfo = ProfilesConfig.getInstance();
```

Comment:
Thanks for the suggestion. I will take a look at it, especially if it will simplify code. For this initial implementation, I continued with what we already use in the code and we have a lot of this `getInstance()` instance everywhere.

Reply:
Okay, it is up to you. But you do not have to continue using this pattern in a new codebase.

I believe, once you end up in a pit, first thing you need to do - stop digging :)